modules/azure/src/main/java/org/testcontainers/containers/CosmosDBEmulatorContainer.java

```diff
@@ -0,0 +1,98 @@
+ */
+public class CosmosDBEmulatorContainer extends GenericContainer<CosmosDBEmulatorContainer> {
+
+    /**
+     * EMULATOR_KEY is a known constant and documented in Azure Cosmos DB Official Pages as well.
+     * Key value is also used as password for emulator certificate file.
+     *
+     * @see <a href="https://docs.microsoft.com/en-us/azure/cosmos-db/local-emulator?tabs=ssl-netstd21#authenticate-requests">Azure Cosmos DB Documents</a>
+     */
+    public static final String EMULATOR_KEY =
+        "C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw==";
+
+    private static final DockerImageName DEFAULT_IMAGE_NAME = parse("mcr.microsoft.com/cosmosdb/linux/azure-cosmos-emulator");
+
+    private static final int PORT = 8081;
+
+    private Path tempDirectory;
+
+    /**
+     * @param dockerImageName specified docker image name to run
+     */
+    public CosmosDBEmulatorContainer(final DockerImageName dockerImageName) {
+        super(dockerImageName);
+        dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+        withExposedPorts(PORT);
+        waitingFor(Wait.forLogMessage("(?s).*Started\\r\\n$", 1));
+    }
+
+    @Override
+    protected void configure() {
+        try {
+            this.tempDirectory = Files.createTempDirectory("azure-cosmosdb-emulator-temp");
+        } catch (Exception ex) {
+            throw new IllegalStateException(ex);
+        }
+    }
+
+    @Override
+    protected void containerIsStarted(InspectContainerResponse containerInfo) {
+        Path keyStorePath = prepareKeyStore();
+        setSystemTrustStoreParameters(keyStorePath.toFile().getAbsolutePath());
+    }
+
+    /**
+     * @return key store path
+     */
+    private Path prepareKeyStore() {
+        String certFileName = "default.sslcert.pfx";
+        Path certFilePath = tempDirectory.resolve(certFileName);
+        copyFileFromContainer("/tmp/cosmos/appdata/" + certFileName, certFilePath.toString());
+        Path keyStorePath = tempDirectory.resolve("cosmos_emulator.keystore");
+        importEmulatorCertificate(certFilePath, keyStorePath);
+        return keyStorePath;
+    }
+
+    private void importEmulatorCertificate(Path pfxLocation, Path keyStoreOutput) {
+        try {
+            KeyStore keystore = KeyStore.getInstance("PKCS12");
+            keystore.load(new FileInputStream(pfxLocation.toFile()), EMULATOR_KEY.toCharArray());
+            keystore.store(new FileOutputStream(keyStoreOutput.toFile()), EMULATOR_KEY.toCharArray());
+        } catch (Exception ex) {
+            throw new IllegalStateException();
+        }
+    }
+
+    /**
+     * @param trustStore keyStore path
+     */
+    private void setSystemTrustStoreParameters(String trustStore) {
+        System.setProperty("javax.net.ssl.trustStore", trustStore);
+        System.setProperty("javax.net.ssl.trustStorePassword", EMULATOR_KEY);
+        System.setProperty("javax.net.ssl.trustStoreType", "PKCS12");
```

Comment:
it would be too dangerous to override the global `trustStore` from a container. I am afraid the best we can do here is to return `KeyStore` to the user, so that the end user decides how to use it with Azure SDK

Reply:
Hmm I see your point.

As a solution; we can show and describe how to set system trust store parameters in test class with documentation markers. Because this is necessary in order to work with azure java client. What do you think?

Reply:
I agree with @bsideup, changing the global `trustStore` as a side effect of starting a container feels very extreme. 
However, documenting how to do it sounds like a very good solution.

Reply:
@okohub sounds good! We could also ask our friends from Microsoft to consider making the SDK a bit more testing friendly (either by not enforcing the SSL or accepting `KeyStore` instances). /cc @JonathanGiles

Reply:
If someone could file an issue in http://GitHub.com/azure/azure-sdk-for-java and ping me, I'll make sure it is discussed internally.

Reply:
@JonathanGiles if you can start the review process of issue I commented above again, would be appreciated :)

Reply:
It looks as if this has already had significant discussion. I do note the comment about less Azure support in testcontainers relative to other cloud platforms, and I would like to see that rectified. I have to admit I don't know anything about testcontainers, but perhaps I can get a crash course at some point so I better understand what is required.

Reply:
@JonathanGiles
please see discussion
https://github.com/Azure/azure-sdk-for-java/pull/22990

Reply:
I have already... That was what my comment above was based on